code/mob/suicide.dm

```diff
@@ -180,14 +180,35 @@
 	if (.)
 		src.visible_message("<span class='alert'><b>[src] suddenly dies for no adequately explained reason!</b></span>")
 
-// instead of dying, flockdrone suicide should hand control back to the mobcritter AI
 /mob/living/critter/flock/drone/do_suicide()
-	emote("beep")
-	say("\[System notification: self-aware drone has voluntarily surrendered its self-awareness and returned to basic function.\]")
-	var/mob/living/C = src.controller
-	src.release_control()
-	if(C)
-		C.suicide()
-		C.unlock_medal("Damned", 1)
-
-	return TRUE
+	if ((locate(/obj/flock_structure/relay) in src.flock.structures) && istype(src.controller, /mob/living/intangible/flock/flockmind) && !length(src.flock.getActiveTraces()))
+		boutput(src, "<span class='alert'>You can't abandon your Flock with the Relay active!</span>")
+		return
+	if (tgui_alert(src, "Are you sure you want to commit suicide?", "Confirm Suicide", list("Yes", "No")) == "Yes")
+		var/mob/living/intangible/flock/C = src.controller
+		src.release_control()
+		if (istype(C, /mob/living/intangible/flock/flockmind))
```

Comment:
Add a `do_suicide` stub proc to `/mob/living/intangible/flock` and override it in flockmind and trace, you can skip this check and just do `C.do_suicide()` instead

Reply:
Idk, they don't really have anything common to them except for a single tgui prompt, would just end up doing an if statement for type and doing what was done in each function.

Reply:
That's fine, by stub proc I just mean an empty proc definition. All intangible flock things are going to need it, and it's better for future code that it's defined in the parent class. I think there's even a flag you can use to force all child classes to implement it.

Reply:
Both do_suicide() functions have a parameter passed in, but if having a stub proc, it wouldn't have one, or if it does it would be unused?

Reply:
You can define it and not use it, no problem :)